src/hotspot/share/gc/shenandoah/shenandoahGenerationalControlThread.cpp

```diff
@@ -45,294 +44,301 @@
 #include "memory/metaspaceUtils.hpp"
 #include "memory/metaspaceStats.hpp"
 #include "runtime/atomic.hpp"
+#include "utilities/events.hpp"
 
 ShenandoahGenerationalControlThread::ShenandoahGenerationalControlThread() :
-  ShenandoahController(),
-  _control_lock(Mutex::nosafepoint - 2, "ShenandoahControlGC_lock", true),
-  _regulator_lock(Mutex::nosafepoint - 2, "ShenandoahRegulatorGC_lock", true),
+  _control_lock(Mutex::nosafepoint - 2, "ShenandoahGCRequest_lock", true),
   _requested_gc_cause(GCCause::_no_gc),
-  _requested_generation(GLOBAL),
-  _degen_point(ShenandoahGC::_degenerated_outside_cycle),
-  _degen_generation(nullptr),
+  _requested_generation(nullptr),
+  _degen_point(ShenandoahGC::_degenerated_unset),
+  _heap(ShenandoahGenerationalHeap::heap()),
+  _age_period(0),
   _mode(none) {
   shenandoah_assert_generational();
   set_name("Shenandoah Control Thread");
   create_and_start();
 }
 
 void ShenandoahGenerationalControlThread::run_service() {
-  ShenandoahGenerationalHeap* const heap = ShenandoahGenerationalHeap::heap();
 
-  const GCMode default_mode = concurrent_normal;
-  ShenandoahGenerationType generation = GLOBAL;
-
-  uint age_period = 0;
+  const int64_t wait_ms = ShenandoahPacing ? ShenandoahControlIntervalMin : 0;
```

Comment:
So we are supporting ShenandoahPacing with GenShenm(at least till we pull it in the future), but don't want to enable it by default, correct?